master/buildbot/test/unit/test_secret_in_file.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import stat
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+from buildbot.secrets.providers.file import SecretInAFile
+from buildbot.test.util.config import ConfigErrorsMixin
+
+
+class TestSecretInFile(ConfigErrorsMixin, unittest.TestCase):
+
+    def createTempDir(self, dirname):
+        tempdir = self.mktemp()
+        basedir = os.path.dirname(tempdir)
+        tempdir = os.path.join(basedir, "temp")
+        os.mkdir(tempdir)
+        return tempdir
+
+    def createFileTemp(self, tempdir, filename, text=None):
+        file_path = os.path.join(tempdir, filename)
+        filetmp = open(file_path, 'w')
+        with open(file_path, 'w') as filetmp:
+            if text:
```

Comment:
it does not make sense to create them without text, it make sense to create them with empty text in that case you make text=''